Gruppe.A1.End-to-End_Encryption_mit_IPv6_und_integriertem_IPSec.ThushjandanPonnudurai.PascalBaumann.Termpaper.tex

```diff
@@ -62,12 +73,103 @@ \chapter{Theoretical part}
 
 \section{IP protocols}
 \label{sec:IPprot}
+The Internet Protocol (IP) is a set of rules governing how packets are transmitted over the Internet IP is probably the common protocol in the internet and is one of the layer 3 protocols (network layer) in the OSI model. It implements two basic functions:
+\begin{itemize}
+	\item Addressing hosts
+	\item Routing datagrams (packets) from a source host to a destination host over multiple IP networks
+\end{itemize}
+A datagram is composed of an IP header and a payload. Source and destination IP addresses and other meta data are parts of the IP header and are required to deliver a datagram. The payload includes data that is transported.
+
+IPv4 is the first major version and is at the moment the dominant protocol version of the Internet. Due to a lack of IPv4 addresses the successor IPv6 was born. \parencite{NadeemUnuth2016}
 
 \subsection{IPv4}
 \label{ssec:IPv4}
+Internet Protocol version 4 (IPv4) is specified in the IETF RFC 791 document and is existing since 1980. To route a packet across the networks, every host in the network has a logical address, which is the IPv4 address. This IPv4 addressing system is based on a 32-bit logical address, which amounts to 4'294'967'296 unique addresses.
+
+An example of an IPv4 address is "148.21.45.110". The address is written in dot decimal notation and it has four octets of 8-bits. The binary form of this address will be 10010100.00010101.00101101.01101110. \parencite{Babatunde2014}
 
+At the moment there is an IPv4 address exhaustion problem. Since the 3th of February 2011 "Internet Corporation for Assigned Names and Numbers" (ICANN) doesn't have any free blocks of IPv4 addresses. Due this problem IPv6 was developed and everything will be migrated in IPv6 over the long-term. 
 \subsection{IPv6}
 \label{ssec:IPv6}
+Internet Protocol version 6 (IPv6) is the successor of IPv4. It was specified in 1998. The growth of the Internet led to a need for new alternatives. The problem is that IPv4 cannot provide the needed number of logical addresses. For that reason IPv6 protocol was developed.\parencite[11]{Babatunde2014}
```

Comment:
[...] led to a _need for alternatives_.
IPv6 is one of the alternatives. Unless there were alternatives to IPv4 before, which you have not yet specified.